doc/user-guide.md

```diff
@@ -11,189 +59,301 @@ funcool/rumext
-the React JSX) for defining the React DOM. The intention of this
-section is to explain only the essential part of it and the
-peculiarities of Rumext.
+You can pass to `:>` macro the name of a custom component (see [below](#creating-a-react-custom-component))
+to create an instance of it:
 
-Let's start with simple generic elements like `div`:
+```clojure
+(mf/defc my-label*
+  [{:keys [name class on-click] :rest props}]
+    [:span {:on-click on-click}
+      [:> :label props name]])
+
+(mf/defc other-component
+  []
+  [:> my-label* {:name "foobar" :on-click some-fn}])
+```
+
+## Creating a React custom component
+
+The `defc` macro is the basic block of a Rumext UI. It's a lightweight utility
+that generates a React **function component** and adds some adaptations for it
+to be more convenient to ClojureScript code, like `camelCase` conversions and
+reserved name changes as explained [above](#passing-props).
+
+For example, this defines a React component:
 
 ```clojure
-[:div {:class "foobar"
-       :style {:background-color "red"}
-       :on-click some-on-click-fn}
-  "Hello World"]
+(require '[rumext.v2 :as mf])
+
+(mf/defc title*
+  [{:keys [label-text] :as props}]
+  [:div {:class "title"} label-text])
 ```
 
-The props and the style are transformed at compile time into a JS
-object, transforming all keys from lisp-case to camelCase (and
-renaming `:class` to `className`); so the compilation results in
-something like this:
+The compiled javascript for this block will be similar to what would be
+obtained for this JSX block:
 
 ```js
-const h = React.createElement;
+export default function title({labelText}) {
+  return (
+    <div className="title">
+      {labelText}
+    </div>
+  );
+}
+```
+
+**NOTE**: the `*` in the component name is a mandatory convention for proper
+visual distinction of React components and Clojure functions. It also enables
+the current defaults on how props are handled. If you don't use the `*` suffix,
+the component will behave in legacy mode (see the [FAQs](#faq) below).
 
-h("div", {className: "foobar",
-          style: {"backgroundColor": "red"},
-          onClick: someFn},
-          "Hello World");
+The component created this way can be mounted onto the DOM:
+
+```clojure
+(ns myname.space
+  (:require
+   [goog.dom :as dom]
+   [rumext.v2 :as mf]))
+
+(def root (mf/create-root (dom/getElement "app")))
+(mf/render! root (mf/element title* #js {:labelText "hello world"}))
 ```
 
-It should be noted that this transformation is only done to properties
-that are keyword types and that properties that begin with `data-` and
-`aria-` are left as-is without transforming, just like string
-keys. The properties can be passed directly using camelCase syntax (as
-React natively expects) if you want.
+Note that when calling `mf/element` you need to give the attributes in the
+raw Javascript form, because this macro does not have automatic conversions.
```

Comment:
Se podria añadir:

Alternatively to the use of `mf/element` we can use the `mf/html` macro such that `(mf/html [:> title* {:label-text "hello-world"])` as a way to create element that obeys the same transformation rules.

Reply:
Done